src/executor/query/DataCollectExecutor.cpp

```diff
@@ -134,13 +140,36 @@ Status DataCollectExecutor::collectMToN(const std::vector<std::string>& vars,
                     ds.rows.emplace_back(seqIter->moveRow());
                 }
             } else {
-                return Status::Error("Iterator should be kind of SequentialIter.");
+                std::stringstream msg;
+                msg << "Iterator should be kind of SequentialIter, but was: " << iter->kind();
+                return Status::Error(msg.str());
             }
             itersHolder.emplace_back(std::move(iter));
         }
     }
     result_.setDataSet(std::move(ds));
     return Status::OK();
 }
+
+Status DataCollectExecutor::collectBFSShortest(const std::vector<std::string>& vars) {
```

Comment:
I think you could call `RowBasedMove` directly.

Reply:
And do we really need to collect the path over history for only shortest path?

Reply:
Will rewrite once we implement returning props for path.